modules: mcuboot: update to the latest

tomi-font opened the pull request:
And pull in needed fixes for the MCUboot update.

tomi-font committed:
modules: trusted-firmware-m: pull in MCUboot fixes

Pull in fixes to update MCUboot.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

sylvioalves commented:
@nordicjm, would you update `hal_espressif`revision to e589b872d92d8515ce512b2d34f255d0bccb0e0f ?

tomi-font committed:
west.yml: update hal_espressif

Update to a newer version to fix CI when updating MCUboot.

Signed-off-by: Tomi Fontanilles <tomi.fontanilles@nordicsemi.no>

nordicjm committed:
west.yml: MCUboot synchronization from upstream

Update Zephyr fork of MCUboot to revision:
  52e2afc2f809c424b0f337f56059d1dfcc7e6d98

Brings following Zephyr relevant fixes:
  - 52e2afc2 boot_serial: Fix `format` warning
  - dd4d6541 boot_serial: Fix `incompatible-pointer-types` warning
  - 2931f9d7 booutil: swap_scratch: Do not check sectors with
    compression
  - feb9265f bootutil: loader: Add protected TLV size to image size
    check
  - 4d85e29f bootutil: loader: Remove images with conflicting flags
  - 206c7e74 bootutil: loader: Remove encrypted/compressed images
    without support
  - 0c1ddf7e zephyr: Add Kconfig for decompression
  - 91d86b8a bootutil: Add compressed image flags and TLV
  - 2939d305 bootutil: loader: Verify image header before checking
    image
  - dbb5c782 bootutil: loader: Add state to boot_is_header_valid()
    function
  - 6071ceb1 boot: Introduce the MCUBOOT_ENC_BUILTIN_KEY option
  - ab4fb329 bootutil: Fix wrong definition of find_swap_count
  - dc030555 imgtool: Add --non-bootable flag
  - aa1f9560 zephyr: prefix generated header path with `zephyr/`
  - 334d2c28 boot: main: avoid unused build warning
  - c87d41ef boot: zephyr: Fix included header
  - 034f58cb bootutil: Fix missing include
  - e2e1e945 bootutil: Fix swap move max app size calculation
  - 4f1ab9e7 bootutil: Add missing images to get maximum image
    size details
  - eaae6500 zephyr: Add support for automatically calculcating
    max sectors
  - 6fe259b1 boot: Simplify copy loop in boot_copy_region
  - 7e69047b boot: zephyr: add support for frdm-mcxn947
  - 899ade2d boot: zephyr: boards: Add mimxrt1062_fmurt6
    configuration
  - ae27bc77 boot: zephyr: boards: Add common mimxrt1050_evk
    configuration.
  - 6123e916 boot: zephyr: boards: Add mimxrt11xx configurations
  - 3f11286e boot: Remove image_index from boot_encrypt
  - 2a7565b3 bootutil: Move encryption key function to common file
  - 0e5e1a9f Reapply "boot: Add MCUBOOT_HW_KEY support for image
    encryption"
  - ef598b13 imgtool: Fix getpub fails for ed25519 key
  - 8456464c zephyr: Fix issue with sysbuild if something else is
    named mcuboot
  - d09112ac boot: Reduce repeating code in
    boot_decrypt_and_copy_image_to_sram
  - 7f9ac979 boot: Change boot_enc_load to take slot number
    instead of image
  - 7fc3ad3d boot: zephyr: Fix RAM load chain load address
  - 49976dc6 imgtool: Bump cryptography library version
  - 33557352 boot: Move encryption context invalidation to
    boot_enc_drop.
  - 2371c0aa boot: Rename boot_enc_decrypt to boot_decrypt_key
  - 4b9d07dc boot: boot_serial: change logging to debug level
  - 7e3a1cec boot: Add missing boot_enc_init
  - 4da4a72c boot: Remove pointless slot identification
  - 08d2d94c bootutil: Keep image encrypted in scratch area
  - f515bb1c bootutil: Properly retrieve image headers after
    interrupted swap-scratch

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
